File: Add Code Here/C++/Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;.
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* ptr1=head;
        ListNode* ptr2=head;
          while(n>0)
            {
                ptr2=ptr2->next;
                n--;
            }
         if(ptr2 ==nullptr)
            return head->next;
        while(ptr2->next!=nullptr)
        {
          
            
             ptr2=ptr2->next;
             ptr1=ptr1->next;
        }
       ptr1->next=ptr1->next->next; 
        return head;
    }
};

File: Add Code Here/JAVA/Java_Medium_Solutions/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
          if (head.next == null) {
            return null;
        }

        //To Find the Size
        int size = 0;
        ListNode curr = head;
        while (curr!= null){
            curr = curr.next;
            size++;
        }

        if (n == size){
            return head.next;
        }

        //To search and Delete
        int index = size - n;
        ListNode prev = head;
        for(int i =1; i < index; i++){
            prev = prev.next;
        }
        prev.next = prev.next.next;  //skipping the node or unlinking the node
        return head;
        
    }
}
